### DISCUSSION TITLE
Fatal error: Path cannot be empty in core.php

### DISCUSSION THREAD

**Discussion by kanbeq** [datenstrom/community#818] [Report a bug] [answered]
Unfortunately, I can't pinpoint exactly when this error occurs, but it does happen at least once every day
PHP error_log:
```
PHP Fatal error:  Uncaught ValueError: Path cannot be empty in /.../system/extensions/core.php:2620
Stack trace:
#0 /.../system/extensions/core.php(2620): fopen()
#1 /.../system/extensions/core.php(242): YellowToolbox->readFile()
#2 /.../system/extensions/core.php(148): YellowCore->sendFile()
#3 /.../system/extensions/core.php(117): YellowCore->processRequest()
#4 /.../yellow.php(9): YellowCore->request()
#5 {main}
  thrown in /.../system/extensions/core.php on line 2620
```
Maybe useful information:
PHP 8.1, installed extensions: blog, command, core, disqus, draft, edit, highlight, image, markdown, meta, polish, private, search, sitemap, toc, update
There is an error in the `yellow-website.log` file as well (`[datetime] error Can't parse file 'system/extensions/core.php'!`).

**Comment by annaesvensson** [comment]
What does the web server access log say? Can you copy/paste the request that's causing the bug?

**Reply by kanbeq** [reply]
This error occurs when the MediaMathbot bot is asking for the sellers.json
file. I tested this page as a client and the error also occurred. There was
a time when the page returned 404 because I don't have such a file.
However, it now returns this error.

I tested some other random addresses that don't exist. They also return this error. A 404 error should be returned

**Reply by annaesvensson** [reply]
Thanks @kanbeq for the detailed bug report, thanks @schulle4u for confirming the bug. This bug was another one that has to do with PHP 8.1 compatibility. In the past PHP internal functions gracefully handled unexpected values, such as empty strings, now many PHP internal function will enforce input parameter validation.  When this happens you will see a `Uncaught TypeError` or `Uncaught ValueError`.

**Comment by kanbeq** [comment, marked as answer]
Works after updating `core.php`. Thanks @markseuffert
